ext/ddtrace.c

```diff
@@ -1185,6 +1191,71 @@ static PHP_FUNCTION(add_distributed_tag) {
     RETURN_NULL();
 }
 
+static PHP_FUNCTION(set_user) {
+    UNUSED(execute_data);
+
+    zend_string *user_id = NULL;
```

Comment:
Can never be NULL after zpp (not optional), you can remove the NULL assign and NULL check later.